gtsfm/utils/pycolmap_utils.py

```diff
@@ -0,0 +1,29 @@
+"""Utility to convert GTSAM types to pycolmap types.
+
+Authors: John Lambert
+"""
+
+import pycolmap
+from gtsam import Cal3Bundler
+
+
+def get_pycolmap_camera(camera_intrinsics: Cal3Bundler) -> pycolmap.Camera:
+    """Convert Cal3Bundler intrinsics to a pycolmap-compatible format (a dictionary).
+
+    See https://colmap.github.io/cameras.html#camera-models for info about the COLMAP camera models.
+    Both SIMPLE_PINHOLE and SIMPLE_RADIAL use 1 focal length.
+    """
+    focal_length = camera_intrinsics.fx()
+    cx, cy = camera_intrinsics.px(), camera_intrinsics.py()
+
+    # TODO (johnwlambert): use more accurate proxy?
+    width = int(cx * 2)
+    height = int(cy * 2)
```

Comment:
I had not noticed this: will this not affect results? Maybe you need more than the Cal3Bundler here?

Reply:
Currently we always set cx,cy as half W,H so it’s identical. We don’t pass the image shape to the verifier, so we can’t read it off directly

Reply:
you mean the loader sets it as half the image size? do we not have the principal point in the intrinsics for some datasets?

Reply:
@akshay-krishnan Good point. That's true that for some datasets we have intrinsics (including principal point) from another source, so image height, image width would not necessarily be 2 * cy, 2 * cx.

However, these values here are just dummy values (image width and image height aren't used anywhere in the F / E / H estimation; rather cx and cy are used in the essential matrix estimation here: https://github.com/colmap/colmap/blob/9f3a75ae9c72188244f2403eb085e51ecf4397a8/src/base/camera_models.h#L629)

So we could technically put any values there. I'll add a comment explaining that.